expfmt/encode.go

```diff
@@ -116,7 +116,13 @@ func NegotiateIncludingOpenMetrics(h http.Header) Format {
 // for FmtOpenMetrics, but a future (breaking) release will add the Close method
 // to the Encoder interface directly. The current version of the Encoder
 // interface is kept for backwards compatibility.
-func NewEncoder(w io.Writer, format Format) Encoder {
+//
+// NewEncoder can be called with additional options to customize the OpenMetrics text output.
+// For example:
+// NewEncoder(w, FmtOpenMetrics_1_0_0, WithCreatedLines())
+//
+// Extra options are ignored for all other formats.
+func NewEncoder(w io.Writer, format Format, options ...ToOpenMetricsOption) Encoder {
```

Comment:
Hm, I think given we can't add additional variadic argument it has be more open, no?

```suggestion
func NewEncoder(w io.Writer, format Format, options ...EncoderOption) Encoder {
```